### DISCUSSION TITLE
OIDC TenantResolver based on annotation presents on REST endpoints

### DISCUSSION THREAD

**Discussion by Eng-Fouad** [quarkusio/quarkus#34663] [Community]
This is a feature request, but instead of opening a new issue, I am not sure if it is possible to be implemented.

Instead of creating custom `TenantResolver`, one can specify the OIDC tenant on JAX-RS resource via annotation, something like:

```
@Tenant("foo")
@Authenticated
@GET
@Path("/abc/def")
public String helloWorld() {
    return "Hello World";
}
```

What do you think? @sberyozkin @pedroigor

**Comment by sberyozkin** [comment]
@Eng-Fouad We have it supported like this:

https://quarkus.io/guides/security-openid-connect-multitenancy#resolving-tenant-identifiers-with-annotations

**Reply by Eng-Fouad** [reply]
Wow, that's great. Thanks for your answer.

**Comment by sberyozkin** [comment]
We can probably simplify it further, so that one can write
```
@Interceptor
@HrTenant
public class HrTenantInterceptor extends TenantInterceptor {
    public  HrTenantInterceptor() {
        super("hr");
    }
}
```

**Comment by sberyozkin** [comment]
@Eng-Fouad Np, `@Tenant("123")` would be ok, but I suspect it would be very complex to do right, we'd have to keep some model in memory, trace what method has been invoked, etc

**Comment by sberyozkin** [comment]
@Eng-Fouad I'll have a look at the suggested simplification, it would be very simple indeed once it is done

**Comment by Eng-Fouad** [comment]
It seems that `HrTenantInterceptor` is called after authentication, hence it never get called as the authentication fails because it gets configuration of wrong tenant :(
